src/main.cpp

```diff
@@ -4656,6 +4656,7 @@ bool static LoadBlockIndexDB()
     const CChainParams& chainparams = Params();
     if (!pblocktree->LoadBlockIndexGuts(InsertBlockIndex, chainparams))
         return false;
+    if (ShutdownRequested()) return true;
```

Comment:
A slight change in the meaning of the return value of this function, `LoadBlockIndexDB()` (and therefore also `LoadBlockIndex()`, which is its only caller): Return value `true` now means there were no errors, instead of "all blocks have been loaded successfully". In other words, if shutdown has been requested, this is not an error, so return `true` (success). I think this makes more sense, although it does require the caller to check if shutdown has been requested.

Reply:
I'd prefer if this returned an enum to represent all the states in which it returns precisely.

Reply:
In https://github.com/bitcoin/bitcoin/pull/12980/files, this returns `false` instead, and the caller then checks fRequestShutdown.

Reply:
I'd prefer taking the approach that upstream did, so the return value's semantics aren't altered. It would also be useful to see if backporting the upstream PR is sufficient, or whether other changes from this PR are needed.

Reply:
ping @LarryRuane